### PR TITLE
Adds formatting script

### PR DESCRIPTION
**Pull request by KipHamiltons**
`./b format` has arrived.
This PR also makes the formatting fixes which is would do when run on the current codebase.

Depends on #39

### CODE REVIEW

File: scripts/format.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        text = text + e.output.decode("utf-8")
+    except:
+        cprint("You don't have one of the formatters installed.\nRun `sudo -H pip3 install -r requirements.txt`", color='red')
+
+    return text
+
+
+def register(command):
+    command.help = "Format all the code in the codebase using clang-format"
+    print("nice kevin")
```
Comment (Bidski) [inline scripts/format.py:40]: Leave kevin alone!

File: scripts/format.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+def run(**kwargs):
+    print("nice one!!")
```
Comment (Bidski) [inline scripts/format.py:43]: Is it though?

File: scripts/format.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import b
+
+
+def _do_format(path):
+    text = ""
+    try:
+        # Use the absolute path to file
+        abs_path = os.path.abspath(path)
+
+        if path.endswith((".h", ".c", ".cc", ".cxx", ".cpp", ".hpp", ".ipp")):
```
Comment (Bidski) [inline scripts/format.py:20]: We should look for `proto` files that are not in the `protos` folder and format them as well

**Review by Bidski — CHANGES_REQUESTED** [review]
You are missing both `.clang-format` and `.cmake-format.py` files

I didn't review `b` and `scripts/b.py` here as they are in #39

File: scripts/format.py (hunk 1 above)
Comment (JosephusPaye) [reply scripts/format.py:40]: Is this a self reference to _Big Kevin_? 😅

Comment (KipHamiltons) [reply scripts/format.py:40]: hahhahaaha anyone who can think of something that doesn't sound stupid for debug print statements is a psychopath

File: scripts/format.py (hunk 2 above)
Comment (KipHamiltons) [reply scripts/format.py:43]: :joy: :joy:

**Comment by KipHamiltons** [comment]
We already have a `.clang-format` and `.cmake-format.py` file in this repo, although the `.clang-format` was slightly out of date, so it had been brought up to parity with the NUbots one.

**Review by Bidski — APPROVED** [review]
LGTM

**Pull request merged**